lib/Doctrine/ORM/Tools/Pagination/LimitSubqueryOutputWalker.php

```diff
@@ -182,41 +211,77 @@ public function walkSelectStatement(SelectStatement $AST)
     }
 
     /**
-     * Generates new SQL for Postgresql or Oracle if necessary.
+     * Finds all PathExpressions in an AST's OrderByClause, and ensures that
+     * the referenced fields are present in the SelectClause of the passed AST.
      *
      * @param SelectStatement $AST
+     */
+    private function addMissingItemsFromOrderByToSelect(SelectStatement $AST)
+    {
+        // This block dumps the order by clause node using Node::dump().
+        // It then finds all PathExpressions within and captures the
+        // identificationVariable and field name of each.
+        $orderByDump = (string)$AST->orderByClause;
+        $selects = [];
+        if (preg_match_all('/PathExpression\([^\)]+"identificationVariable": \'([^\']*)\'[^\)]+"field": \'([^\']*)\'[^\)]+\)/i', $orderByDump, $matches, PREG_SET_ORDER)) {
+            foreach($matches as $match) {
+                $idVar = $match[1];
+                $field = $match[2];
+                if (!isset($selects[$idVar])) {
+                    $selects[$idVar] = [];
+                }
+                $selects[$idVar][$field] = true;
+            }
+        }
+
+        // Loop the select clause of the AST and exclude items from $select
+        // that are already being selected.
+        foreach ($AST->selectClause->selectExpressions as $selectExpression) {
+            if ($selectExpression instanceof SelectExpression) {
```

Comment:
I suppose it's a sanity check because of public property? Maybe not necessary as the AST is internal API I suppose...